### ISSUE TITLE
Can't get it to work

### ISSUE THREAD

**Issue by thedaryen** [ArthurVaiselbuh/HeadsetUtils#2]
I've followed instructions, but the program doesn't seem to swap audio devices automatically when I turn my headset on/off.

Here's the audio devices:

![image](https://user-images.githubusercontent.com/20780420/216833054-4a6e84ed-416c-4dc7-a9f9-c37a82872cf2.png)

Here's my app.config

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
	<appSettings>
		<add key="MonitoringIntervalMs" value="3000"/>
		<add key="HeadsetName" value="HS80"/>
		<add key="SpeakersName" value="Edifier"/>
		<add key="Autorun" value="true"/>
	</appSettings>
</configuration>
```

The Powershell cmdlet is installed, I've tested it through Poweshell and it's working. HeadsetUtils is running in background, I can see it on Task Manager. But when I turn the headset on and off, nothing happens, the default audio device stays the same.

I'm on Windows 11 btw if it makes any difference.

**Comment by ArthurVaiselbuh** [comment]
@andreycout 
First try modifying `HeadsetUtils.dll.config` and set `<add key="HeadsetName" value="HS80"/>` and your other changes there(restart the program after modifying the file). 
If that doesn't work please take the following steps:
1. Change logging level to DEBUG via log4net.config:
`<level value="DEBUG" />`
2. restart the HeadsetUtils program 
3. Shut off the headset and turn it back on, and note the time
4. attach the `HeadsetUtils.log` file (should be next to the EXE) and the relevant log file from `%LOCALAPPDATA%\Corsair\CUE4\logs` to this ticket, and write down the time from step 4.

**Comment by thedaryen** [comment]
[2023-02-05T13-11-54.log](https://github.com/ArthurVaiselbuh/HeadsetUtils/files/10613765/2023-02-05T13-11-54.log)

17:53

**Comment by ArthurVaiselbuh** [comment]
Hmm looks like your corsair log file doesn't have the same messages that mine does when the device is turned on or off.
Take a look at [LogFileConnectionEventSource](https://github.com/ArthurVaiselbuh/HeadsetUtils/blob/master/HeadsetUtils/LogFileConnectionEventSource.cs), my log files have lines that match the regex `Process isConnected: ([a-z]+) for ([^\\(]+)`
If there are specific log messages you can recognize when your device is turned on or off I guess we could create separete regexes for them and add them to the appconfig

**Comment by thedaryen** [comment]
> 
Checking my log, I can identify the following messages:
When the device is turned off, I get:
`2023-02-06 18:06:41.053 I cue.devices.audio.comp: Audio control device was cleared`
When the device is turned on, I get:
`2023-02-06 17:43:12.487 I cue.devices: Device insertion processing complete for HS80 DONGLE (vid=1b1c, pid=a6b)`

**Comment by ArthurVaiselbuh** [comment]
@andreycout  I released[ version 1.0.2 ](https://github.com/ArthurVaiselbuh/HeadsetUtils/releases/tag/1.0.2)see if it works for you by overriding the appconfigs `ConnectedRegexOverride` and `DisconnectedRegexOverride`

**Comment by thedaryen** [comment]
It works now! Thanks a lot!
